src/server/scripts/Outland/TempestKeep/arcatraz/boss_harbinger_skyriss.cpp

```diff
@@ -15,147 +15,148 @@
-        MindRend_Timer = 3000;
-        Fear_Timer = 15000;
-        Domination_Timer = 30000;
-        ManaBurn_Timer = 25000;
     }
 
-    bool Intro;
-    bool IsImage33;
-    bool IsImage66;
-
     uint32 Intro_Phase;
     uint32 Intro_Timer;
-    uint32 MindRend_Timer;
-    uint32 Fear_Timer;
-    uint32 Domination_Timer;
-    uint32 ManaBurn_Timer;
 
     void Reset() override
     {
+        DoCastSelf(SPELL_SIMPLE_TELEPORT);
         _Reset();
-
-        me->SetImmuneToAll(!Intro);
+        _intro = false;
+        _image33 = false;
+        _image66 = false;
+        me->SetImmuneToAll(!_intro);
         Initialize();
     }
 
-    void MoveInLineOfSight(Unit* who) override
+    void JustEngagedWith(Unit* who) override
     {
-        if (!Intro)
-            return;
-
-        ScriptedAI::MoveInLineOfSight(who);
-    }
-
-    void JustEngagedWith(Unit* /*who*/) override { }
-
-    void JustDied(Unit* /*killer*/) override
-    {
-        Talk(SAY_DEATH);
-        _JustDied();
+        BossAI::JustEngagedWith(who);
+        events.ScheduleEvent(EVENT_MIND_REND, 2s, 10s);
+        events.ScheduleEvent(EVENT_FEAR, 10s, 20s);
+        events.ScheduleEvent(EVENT_DOMINATION, 30s, 40s);
+        if (IsHeroic())
+            events.ScheduleEvent(EVENT_MANA_BURN, 25s);
     }
 
-    void JustSummoned(Creature* summon) override
+    void OnSpellCast(SpellInfo const* spell) override
     {
-        if (!summon)
-            return;
-        if (IsImage66)
-            summon->SetHealth(summon->CountPctFromMaxHealth(33));
-        else
-            summon->SetHealth(summon->CountPctFromMaxHealth(66));
-        if (me->GetVictim())
-            if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0))
-                summon->AI()->AttackStart(target);
-
-        summons.Summon(summon);
-        summon->SetImmuneToAll(false);
+        switch (spell->Id)
+        {
+            case SPELL_FEAR:
+                Talk(SAY_FEAR);
+                break;
+            case SPELL_DOMINATION:
+            case SPELL_DOMINATION_H:
+                Talk(SAY_MIND);
+                break;
+            default:
+                break;
+        }
     }
 
     void KilledUnit(Unit* victim) override
     {
-        //won't yell killing pet/other unit
+        // Won't yell killing pet/other unit
         if (victim->GetEntry() == NPC_ALPHA_POD_TARGET)
             return;
 
         Talk(SAY_KILL);
     }
 
-    void DoSplit(uint32 val)
+    void DamageTaken(Unit* /*killer*/, uint32& damage, DamageEffectType /*damageType*/, SpellInfo const* /*spellInfo = nullptr*/) override
     {
-        if (me->IsNonMeleeSpellCast(false))
-            me->InterruptNonMeleeSpells(false);
-
-        Talk(SAY_IMAGE);
+        if (me->HealthBelowPctDamaged(66, damage) && !_image66)
+        {
+            _image66 = true;
+            events.ScheduleEvent(EVENT_SUMMON_66, 0s);
+        }
+        if (me->HealthBelowPctDamaged(33, damage) && !_image33)
```

Comment:
```suggestion
        if (!_image33 && me->HealthBelowPctDamaged(33, damage))
```

should it be possible that both summons happen at the same time? if not use `else if`